puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/vars/foreman/2.0.groovy

```diff
@@ -17,5 +17,9 @@ def pipelines = [
         'centos7',
         'debian10',
         'ubuntu1804'
+    ],
+    'upgrade': [
+        'centos7',
+        'ubuntu1804'
```

Comment:
no debian10 here, as upgrading to 2.0 means 1.23→1.24→2.0 and there is no 1.23 on debian10.

Reply:
I thought https://github.com/theforeman/forklift/blob/cee201325469f2228195068fda7ea0be68048313/roles/forklift_versions/lookup_plugins/forklift_versions.py#L97-L98 dealt with this by duplicating the first entry until it's there. So it'll test 1.24 -> 1.24 -> 2.0. https://github.com/theforeman/forklift/blob/cee201325469f2228195068fda7ea0be68048313/pipelines/upgrade/05-server_to_intermediate.yaml#L21 then skips 1.24 -> 1.24 upgrade.

So as long as there's n-1, it's valid to list.

Reply:
This handles the case when the previous version is not defined at all. In this case it's defined, but just not for that specific OS, and that's something `forklift_versions` does not check.

Reply:
Perhaps we can add that check by passing in the OS to the lookup?

Reply:
bleh, more work ;-)

Reply:
https://github.com/theforeman/forklift/pull/1172